products/messaging/frontend/Campaigns/Campaign.tsx

```diff
@@ -0,0 +1,90 @@
+import '@xyflow/react/dist/style.css'
+
+import { LemonTabs } from '@posthog/lemon-ui'
+import { useValues } from 'kea'
+import { Form } from 'kea-forms'
+import { router } from 'kea-router'
+import { PageHeader } from 'lib/components/PageHeader'
+import { LemonField } from 'lib/lemon-ui/LemonField'
+import { LemonInput } from 'lib/lemon-ui/LemonInput'
+import { capitalizeFirstLetter } from 'lib/utils'
+import { SceneExport } from 'scenes/sceneTypes'
+import { urls } from 'scenes/urls'
+
+import { campaignLogic, CampaignLogicProps, CampaignTab, CampaignTabs } from './campaignLogic'
+import { WorkflowEditor } from './Workflows/WorkflowEditor'
+
+export const scene: SceneExport = {
+    component: Campaign,
+    logic: campaignLogic,
+    paramsToProps: ({ params: { id } }): CampaignLogicProps => ({
+        id,
+    }),
+}
+
+export function Campaign({ id }: CampaignLogicProps = {}): JSX.Element {
+    const { currentTab } = useValues(campaignLogic)
+    const { campaignLoading } = useValues(campaignLogic)
+
+    const isNewCampaign = id === 'new'
+
+    const tabContent = {
```

Comment:
Generally I'd highly recommend stealing most of the [HogFunctionScene](https://github.com/PostHog/posthog/blob/master/frontend/src/scenes/hog-functions/HogFunctionScene.tsx) here as you'll get a way better basis for routing which we will inevitably forget about later and then have the pain of retrofitting